docs/important/relevant_functionalities_de.md

```diff
@@ -0,0 +1,1250 @@
+- Dogu-Abfragen
+  - `.Dogus.IsEnabled <Dogu-Name>` - gibt einen `bool`-Wert zurück, ob ein Dogu installiert ist
+  - `.Dogus.Get <Dogu-Name>` - gibt ein [Dogu](../core/compendium_de.md#type-dogu)-Struct zurück
+  - `.Dogus.GetAll` - gibt ein Slice aller [Dogus](../core/compendium_de.md#type-dogu) zurück
+
+Beispiel einer Template-Datei:
+
+```gotemplate
+# my-pseudo-config.conf
+[Global]
+admin.username = {{ .Env.Get "ADMIN_USERNAME" }}
+funny.name = {{ .Config.Get "something_funny" }}
+log.level = {{ .Config.GetOrDefault "log_level" "WARN" }}
+url.jdbc = jdbc:postgresql://postgresql:5432/{{ .Config.GetAndDecrypt "sa-postgresql/database" }}
+url.fqdn = https://{{ .GlobalConfig.Get "fqdn" }}/my-dogu 
+
+{{ if .Config.Exists "notification" }}
+notification = mail
+{{ end }}
+
+{{ if .Dogus.IsEnabled "redmine" }}
+ticketsystem.url = https://{{ .GlobalConfig.Get "fqdn" }}/redmine
+{{ end }}
+...
+```
+
+#### doguctl wait-for-tcp
+
+Mit diesem Aufruf wartet `doguctl` bis ein gegebener TCP-Port offen ist oder ein Timeout (in Sekunden) eintritt. Ein Exit Code != 0 signalisiert einen Fehler.
+
+```bash
+if ! doguctl wait-for-tcp --timeout 120 8080; then
+  echo "Received timeout during wait for port 8080. Exiting."
+  exit 1
+fi
+```
+
+#### doguctl wait-for-http
+
+Mit diesem Aufruf wartet `doguctl` bis eine gegebene HTTP-URL bereit ist oder ein Timeout (in Sekunden) eintritt. Ein Exit Code != 0 signalisiert einen Fehler.
+
+```bash
+if ! doguctl wait-for-tcp --host postgresql --timeout 120 5432; then
+  echo "Received timeout during wait for port 5432 of host postgresql. Exiting."
+  exit 1
+fi
+```
+
+## Service Accounts
+
+Service Accounts bilden einen Mechanismus für Zugangskonten ab, die Dogus zur Absicherung oder Speicherung ihrer Daten benötigen, diese Funktionalität aber nicht selbst bereitstellen möchten. Dogus können sich als Produzent und/oder als Konsument von Service Accounts darstellen. Bei beiden handelt es sich um ein jeweils optionales Feature eines Dogus. 
+
+In der Regel diktiert die Natur der Applikation im Dogu selbst, ob bzw. welches von beidem am sinnvollsten ist. So wäre ein Datenbankmanagement-Dogu wie PostgreSQL nicht hilfreich, das anderen Dogus keine Zugänge zu seiner Datenbasis anbietet. Andererseits wäre es für ein Dogu wie Redmine fatal, kein Dogu zu haben, in das es per SQL seine Daten ablegen kann.
+
+Dogus, die anderen Dogus gegenüber Zugangskonten bereitstellen können, nennt man auch Service-Account-Produzent. Dogus, die ein Zugangskonto bei einem anderen Dogu benötigen, nennt man Service-Account-Konsument.
+
+### Service Accounts produzieren 
+
+Service Accounts werden nicht von einem Konsumer-Dogu selbst angefragt, weil ein Dogu sich nicht um diese Form der Orchestrierung kümmern sollte. Dies ist stattdessen die Aufgabe von einem Client wie `cesapp` oder `k8s-dogu-operator` während der Installation des Service-Account-Konsument-Dogus. Das `ExposedCommand` hierfür lautet `service-account-create`. 
```

Comment:
Kurz zuvor eingeführte Begriffe werden hier nicht genutzt.
Sind Service-Account eigentlich nur für Dogus oder später auch Operators o.ä.? In Kubernetes gibt es ja automatische Service-Bindings z.B. für Datenbanken für sowas.
```suggestion
Service Accounts werden nicht von einem Service-Account-Konsumenten selbst angefragt, weil ein Dogu sich nicht um diese Form der Orchestrierung kümmern sollte. Dies ist stattdessen die Aufgabe von einem Client wie `cesapp` oder `k8s-dogu-operator` während der Installation des Service-Account-Konsumenten. Der `ExposedCommand` hierfür lautet `service-account-create`. 
```

Reply:
done